src/names/applications.cpp

```diff
@@ -183,4 +185,29 @@ IsValidJSONOrEmptyString (const std::string& text){
     UniValue v;
 
     return text.empty() || v.read(text);
+}    
+
+bool
+IsMinimalJSONOrEmptyString (const std::string& text){
+    UniValue v;
+    if(text.empty()){
+        return true;
+    } else if(!v.read(text)){ 
```

Comment:
Since you `return` in the `if` clause, there is no need for `else`, and I think this makes the code more confusing.  Please just make this "if ... return; if ... return; ...".